src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -668,6 +671,13 @@ class Address {
     }
     return false;
   }
+
+  #define create_imm_offset(klass, field_offset_func) \
+  ([]() { \
+    constexpr size_t max_possible_offset = sizeof(klass); \
+    static_assert(Address::offset_ok_for_immed_compile_time<max_possible_offset, 0>(), "must fit in instruction"); \
+    return klass::field_offset_func(); \
+  }())
```

Comment:
This could be done without a Lambda, which subjectively would be a bit simpler, by using the comma operator like this:

```
  template <class K> static void class_size_check() {
    constexpr size_t max_possible_offset = sizeof(K); \
    static_assert(Address::offset_ok_for_immed_compile_time<max_possible_offset, 0>(), "must fit in instruction"); \
  }
  #define create_imm_offset(klass, field_offset_func) \
    (Address::class_size_check<klass>(), klass::field_offset_func())
```